app/gui/src/dashboard/components/AriaComponents/Form/components/FormProvider.tsx

```diff
@@ -39,6 +39,7 @@ export function useFormContext<Schema extends types.TSchema>(
   if (form != null && 'control' in form) {
     return form
   } else {
+    // eslint-disable-next-line react-compiler/react-compiler
```

Comment:
i think we should add comments for all such lint disables. i assume this one is conditional calling given the `rules-of-hooks` ignore but still worth adding a comment imo.

(alternative we should each branch up into its own component?)

Reply:
well, there's a thing that it's safe to call `useContext` conditionally and `React` allows that, though it's not documented anywhere 🌚

In react@19 there'll be a `use` function that can read context values and doesn't violate rules of hooks when called conditionally.